### ISSUE TITLE
Auto detect used C# libraries

### ISSUE THREAD

**Issue by abitrolly** [dense-analysis/ale#3668]
When using `mcs` on Fedora, ALE doesn't detect used assemblies and requires them to be added manually, such as `System.Net.Http` in the example below. Is would be nice if ALE could detect those and do not fail with the error.

```csharp
  using System;
  using System.Net.Http;
  
  class Program
  {
>>    static HttpClient Client = new HttpClient();
  
      static void Main(string[] args)
      {
  
          var hyper = "inter";
          Console.WriteLine($"string {hyper}polation");
      }
  }
~                                                                                                                          
~                                                                                                                          
~                                                                                                                          
CS0246: The type or namespace name `HttpClient' could not be found. Are you missing an assembly reference?
```

**Comment by cprn** [comment]
> ALE doesn't detect used assemblies and requires them to be added manually

How exactly do you add assemblies manually? I just started learning C# and decided using Godot for that purpose would be fun and, well, it isn't because:

```
CS0246: The type or namespace name `Godot' could not be found. Are you missing an assembly reference?
```

AFAIU assembly is in the solution file for the project (`csharp-ls` read it without issues).

**Comment by LandonDeam** [comment]
Same issue occurred for me in nvim with it throwing the false error `CS0518: Predefined type 'System.Runtime.CompilerServices.NullableContextAttribute' is not defined or imported` in a fairly standard .NET 7.0 project. It's been almost 2 and a half years with no out-of -the-box fix now, so I hope this gets addressed at some point to make the solution easier to find, even if it's just stating for people using mono in their projects to use the above lines in their `.vimrc`
